rs/utils/src/mem.rs

```diff
@@ -39,12 +41,25 @@ pub fn u128s_to_lows_highs(ids: &[u128]) -> LowsAndHighs {
     result
 }
 
-pub fn lows_and_highs_to_u128s(lows: &[u64], highs: &[u64]) -> Vec<u128> {
-    let mut result = Vec::with_capacity(lows.len());
+pub fn bytes_to_u128s(bytes: &Vec<u8>) -> Vec<u128> {
+    let chunks = bytes.chunks_exact(16);
+    chunks
+        .map(|chunk| {
+            let mut value: u128 = 0;
+            for (i, &byte) in chunk.iter().enumerate() {
+                value |= (byte as u128) << (i * 8);
+            }
+            value
+        })
+        .collect()
```

Comment:
just combine into a single statement
```suggestion
    bytes.chunks_exact(16)
        .map(|chunk| {
            let mut value: u128 = 0;
            for (i, &byte) in chunk.iter().enumerate() {
                value |= (byte as u128) << (i * 8);
            }
            value
        })
        .collect()
```